### DISCUSSION TITLE
[Question] Change observable several times inside Action

### DISCUSSION THREAD

**Discussion by subzero911** [mobxjs/mobx.dart#762] [General]
An `Action` batches all Observable changes and gives us a final result only.
What if I want not to batch changes, and want to rebuild Observer content on every Observable change?
Example - an optimistic update scenario:
- I click a "Like" button
- I change the Observable and a button turns red
- I perform a web request
- If it fails, I change the Observable again and button turns back to transparent

It all comes within a single Action.
It's easily achieved with notifyListeners(), but i don't how to do it with MobX.

**Comment by amondnet** [comment]
@subzero911 I haven't tested it, but how about doing something like this:

```dart
@action 
void  foo() {
  //  I change the Observable and a button turns red
}

@action 
void  bar() {
  //  I change the Observable again and button turns back to transparent
}

// without @action
Future<void> optimisticUpdate() {
  try {
    // I change the Observable and a button turns red
    foo();
    // I perform a web request
    performWebRequest();
  } catch (e) {
    // If it fails, I change the Observable again and button turns back to transparent
    bar();
  }
}
```

**Reply by Dalakoti07** [reply]
@amondnet Its is not working I tried this

**Reply by amondnet** [reply]
```dart
import 'dart:math';

import 'package:mobx/mobx.dart';

// Include generated file
part 'counter.g.dart';

class Counter = _Counter with _$Counter;

abstract class _Counter with Store {
  @readonly
  int _value = 0;

  @readonly
  bool _loading = false;

  Future<void> makeApiCall() async{
    _loading = true;
    // simulating api call
    await Future<dynamic>.delayed(const Duration(milliseconds: 1000));
    _loading = false;
  }

  Future<void> increment() async {
    _value = 0;
    await Future<dynamic>.delayed(const Duration(milliseconds: 1000));
    var _random = new Random();
    var _diceface = _random.nextInt(6) +1 ;
    _value = _diceface;
  }
}

```


![untitled](https://user-images.githubusercontent.com/1964421/173802801-eb7a92d2-5739-40c2-97c6-01e67e9eb557.gif)

**Reply by Dalakoti07** [reply]
@amondnet can you tell flutter version, dart version

and mob dependencies versions, and development machine

**Reply by amondnet** [reply]
@Dalakoti07  https://github.com/amondnet/mobx-issue-832/commit/8673b93f60b89fef3df8ca6c911aeb9445e12d8d

I deleted two lines from your code and it works.

**Reply by Dalakoti07** [reply]
Thanks, its working, but in docs its mentioned that @action is used 

https://mobx.netlify.app/api/action

is @action only for isolation/atomic changes?
